planning/behavior_path_planner/include/behavior_path_planner/occupancy_grid_map/occupancy_grid_map.hpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BEHAVIOR_PATH_PLANNER__OCCUPANCY_GRID_MAP__OCCUPANCY_GRID_MAP_HPP_
+#define BEHAVIOR_PATH_PLANNER__OCCUPANCY_GRID_MAP__OCCUPANCY_GRID_MAP_HPP_
+
+#include <tier4_autoware_utils/geometry/geometry.hpp>
+
+#include <autoware_auto_planning_msgs/msg/path_with_lane_id.hpp>
+#include <geometry_msgs/msg/pose_array.hpp>
+#include <nav_msgs/msg/occupancy_grid.hpp>
+
+#include <tf2/utils.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+
+#include <vector>
+
+namespace behavior_path_planner
+{
+int discretizeAngle(const double theta, const int theta_size);
+
+struct IndexXYT
+{
+  int x;
+  int y;
+  int theta;
+};
+
+struct IndexXY
+{
+  int x;
+  int y;
+};
+
+IndexXYT pose2index(
+  const nav_msgs::msg::OccupancyGrid & costmap, const geometry_msgs::msg::Pose & pose_local,
+  const int theta_size);
+
+geometry_msgs::msg::Pose index2pose(
+  const nav_msgs::msg::OccupancyGrid & costmap, const IndexXYT & index, const int theta_size);
+
+geometry_msgs::msg::Pose global2local(
+  const nav_msgs::msg::OccupancyGrid & costmap, const geometry_msgs::msg::Pose & pose_global);
+
+geometry_msgs::msg::Pose local2global(
+  const nav_msgs::msg::OccupancyGrid & costmap, const geometry_msgs::msg::Pose & pose_local);
+
+struct VehicleShape
```

Comment:
IMO
You had better use parameter from vehicle info instead.

Reply:
I think the vehile_info is already used.
https://github.com/autowarefoundation/autoware.universe/pull/873

Reply:
yes, here in pull_over_module.cpp.We can add margin for collision check.

![image](https://user-images.githubusercontent.com/39142679/177666352-ef8000fb-37f6-4d2e-bd22-61128145471f.png)